### PR TITLE
WINDUP-578 - Link Identified Archives to Maven Central by SHA1 Hash

### PR DESCRIPTION
**Pull request by bradsdavis**
Link Identified Archives to Maven Central by SHA1 Hash:
Done through the Maven Central search API & the SHA1 Hash on the archive.

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/712/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/84/
Test PASSed.

**Comment by lincolnthree** [comment]
Merged.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/freemarker/ProjectModelSha1Archive.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import org.jboss.windup.rules.apps.java.archives.model.IdentifiedArchiveModel;
+import org.jboss.windup.util.ExecutionStatistics;
+import org.jboss.windup.util.Logging;
+
+import freemarker.ext.beans.StringModel;
+import freemarker.template.TemplateModelException;
+
+/**
+ * 
+ * Given a {@link ProjectModel}, return the {@link ArchiveModel} that is associated with the application.
```
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/freemarker/ProjectModelSha1Archive.java]: It appears to actually return a SHA1 hash?

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/freemarker/ProjectModelSha1Archive.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    @Override
+    public String getMethodName()
+    {
+        return NAME;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Takes a parameter of type " + ProjectModel.class.getSimpleName() + " and returns the associated "
```
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/freemarker/ProjectModelSha1Archive.java]: This description seems to be incorrect.

File: rules-java/api/src/main/resources/reports/templates/java_application.ftl [hunk 3]
```diff
@@ -102,6 +102,12 @@
 							<#if projectModel.url?has_content>
 								<a href="${projectModel.url}">Project Site</a>
 							</#if>
+									
+							<#if projectModelSha1Archive(projectModel)?has_content>
+								<#assign sha1URL = '|ga|1|1:"' + projectModelSha1Archive(projectModel) + '"'>
+								<#assign sha1URL = 'http://search.maven.org/#search' + sha1URL?url('ISO-8859-1')>
```
Comment (jsight) [inline rules-java/api/src/main/resources/reports/templates/java_application.ftl]: Good. :)

**Pull request closed without merging**